File: src/RPSKata/RPS.java
package RPSKata;

import static RPSKata.Result.P1_WINS;
import static RPSKata.Result.P2_WINS;
import static RPSKata.Throw.*;

class RPS {
    static Result play(Throw p1, Throw p2) {
        if (p1 == ROCK) {
            return P1_WINS;
        }

        return P2_WINS;
    }
}

File: swift/RPS/Sources/RPS/RPS.swift
enum Throw {
    case rock, paper, scissors
}

enum GameResult {
    case p1wins, p2wins, draw
}

public struct RPS {
    func play(p1: Throw, p2: Throw) -> GameResult {
        if (p1 == p2) {
            return .draw
        }

        if  (p1 == .rock && p2 == .paper) ||
            (p1 == .paper && p2 == .scissors) ||
            (p1 == .scissors && p2 == .rock) {
            return .p2wins
        }

        return .p1wins
    }
}
